node/src/components/contract_runtime.rs

```diff
@@ -480,8 +486,9 @@ impl ContractRuntime {
                         );
                         exec_queue
                             .lock()
-                            .unwrap()
+                            .expect("components::contract_runtime: couldn't enqueue block for execution; mutex poisoned")
                             .insert(finalized_block_height, (finalized_block, deploys));
+                        self.metrics.exec_queue_size.inc();
```

Comment:
Technically, `exec_queue` is a `BTreeMap`, hence `insert()` may not actually insert anything, but replace the previous value. So... should this say
```suggestion
                        self.metrics.exec_queue_size.set(self.queue_depth() as i64);
```

Reply:
I guess it would always insert in this case since it has a height bigger than anything it found so far, but I agree to make it a bit more robust and update based on the true size once. Fixed in https://github.com/casper-network/casper-node/pull/3471/commits/dd63ed6169d291f65297d53fc760f5bf086eabe7